collect_app/src/main/java/org/odk/collect/android/widgets/WidgetAnswerFactory.kt

```diff
@@ -0,0 +1,52 @@
+package org.odk.collect.android.widgets
+
+import android.content.Context
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.lifecycle.viewmodel.compose.viewModel
+import org.javarosa.core.model.Constants
+import org.javarosa.form.api.FormEntryPrompt
+import org.odk.collect.android.utilities.MediaUtils
+import org.odk.collect.android.utilities.QuestionMediaManager
+import org.odk.collect.android.widgets.video.VideoWidgetAnswer
+import org.odk.collect.android.widgets.video.VideoWidgetAnswerViewModel
+
+@Composable
+fun widgetAnswer(
+    prompt: FormEntryPrompt,
+    context: Context,
+    answer: String?,
+    questionMediaManager: QuestionMediaManager,
+    mediaUtils: MediaUtils
+): (@Composable () -> Unit)? = when (prompt.controlType) {
+    Constants.CONTROL_VIDEO_CAPTURE -> videoWidgetAnswer(context, answer, questionMediaManager, mediaUtils)
+    else -> throw IllegalArgumentException("Unsupported control type: ${prompt.controlType}")
+}
+
+@Composable
+private fun videoWidgetAnswer(
+    context: Context,
+    answer: String?,
+    questionMediaManager: QuestionMediaManager,
+    mediaUtils: MediaUtils
+): (@Composable () -> Unit)? {
+    val file = questionMediaManager.getAnswerFile(answer) ?: return null
+
+    val viewModel: VideoWidgetAnswerViewModel = viewModel(
+        factory = VideoWidgetAnswerViewModel.Factory(questionMediaManager, mediaUtils)
+    )
+
+    val bitmapFlow = remember(answer) {
+        viewModel.init(answer, context)
+    }
+
+    return {
+        val bitmap by bitmapFlow.collectAsState()
```

Comment:
`collectAsState` [is not lifecycle aware](https://developer.android.com/develop/ui/compose/state#use-other-types-of-state-in-jetpack-compose) which can cause recompositions at inappropriate times I think. I've found that doing `asLiveData().observeAsState()` to a succinct way to using `StateFlow` safely in Compose.